### PR TITLE
[DOCS] Edit snapshot and restore summaries and descriptions

### PR DESCRIPTION
**Pull request by lcawl**
Relates to https://github.com/elastic/elasticsearch-specification/issues/3226

This PR edits the snapshot and restore APIs (https://www.elastic.co/docs/api/doc/elasticsearch/group/endpoint-snapshot) based on information from https://www.elastic.co/guide/en/elasticsearch/reference/master/snapshot-restore-apis.html

### CODE REVIEW

File: output/openapi/elasticsearch-openapi.json [hunk 1]
```diff
@@ -32220,7 +32243,11 @@
...
+        "description": "Restore a snapshot of a cluster or data streams and indices.\n\nYou can only restore a snapshot to a running cluster with an elected master node.\nThe snapshot repository must be registered and available to the cluster.\nThe snapshot and cluster versions must be compatible.\n\nTo restore a snapshot, the cluster's global metadata must be writable. Ensure there are't any cluster blocks that prevent writes. The restore operation ignores index blocks.\n\nBefore you restore a data stream, ensure the cluster contains a matching index template with data streams enabled. To check, use the index management feature in Kibana or the get index template API:\n\n```\nGET _index_template/*?filter_path=index_templates.name,index_templates.index_template.index_patterns,index_templates.index_template.data_stream\n```\n\nIf no such template exists, you can create one or restore a cluster state that contains one. Without a matching index template, a data stream can't roll over or create backing indices.\n\nIf your snapshot contains data from App Search or Workplace Search, you must restore the Enterprise Search encryption key before you restore the snapshot.",
```
Comment (karenzone) [inline output/openapi/elasticsearch-openapi.json:32247]:
```suggestion
        "description": "Restore a snapshot of a cluster or data streams and indices.\n\nYou can  restore a snapshot only to a running cluster with an elected master node.\nThe snapshot repository must be registered and available to the cluster.\nThe snapshot and cluster versions must be compatible.\n\nTo restore a snapshot, the cluster's global metadata must be writable. Ensure there are't any cluster blocks that prevent writes. The restore operation ignores index blocks.\n\nBefore you restore a data stream, ensure the cluster contains a matching index template with data streams enabled. To check, use the index management feature in Kibana or the get index template API:\n\n```\nGET _index_template/*?filter_path=index_templates.name,index_templates.index_template.index_patterns,index_templates.index_template.data_stream\n```\n\nIf no such template exists, you can create one or restore a cluster state that contains one. Without a matching index template, a data stream can't roll over or create backing indices.\n\nIf your snapshot contains data from App Search or Workplace Search, you must restore the Enterprise Search encryption key before you restore the snapshot.",
```

File: output/openapi/elasticsearch-openapi.json [hunk 2]
```diff
@@ -32341,7 +32368,8 @@
         "tags": [
           "snapshot"
         ],
-        "summary": "Returns information about the status of a snapshot",
+        "summary": "Get the snapshot status",
+        "description": "Get a detailed description of the current state for each shard participating in the snapshot.\nNote that this API should only be used to obtain detailed shard-level information for ongoing snapshots.\nIf this detail is not needed or you want to obtain information about one or more existing snapshots, use the get snapshot API.\n\nWARNING: Using the API to return the status of any snapshots other than currently running snapshots can be expensive.\nThe API requires a read from the repository for each shard in each snapshot.\nFor example, if you have 100 snapshots with 1,000 shards each, an API request that includes all snapshots will require 100,000 reads (100 snapshots x 1,000 shards).\n\nDepending on the latency of your storage, such requests can take an extremely long time to return results.\nThese requests can also tax machine resources and, when using cloud storage, incur high processing costs.",
```
Comment (karenzone) [inline output/openapi/elasticsearch-openapi.json:32372]:
```suggestion
        "description": "Get a detailed description of the current state for each shard participating in the snapshot.\nNote that this API should be used only to obtain detailed shard-level information for ongoing snapshots.\nIf this detail is not needed or you want to obtain information about one or more existing snapshots, use the get snapshot API.\n\nWARNING: Using the API to return the status of any snapshots other than currently running snapshots can be expensive.\nThe API requires a read from the repository for each shard in each snapshot.\nFor example, if you have 100 snapshots with 1,000 shards each, an API request that includes all snapshots will require 100,000 reads (100 snapshots x 1,000 shards).\n\nDepending on the latency of your storage, such requests can take an extremely long time to return results.\nThese requests can also tax machine resources and, when using cloud storage, incur high processing costs.",
```

File: output/schema/schema.json [hunk 3]
```diff
@@ -18330,9 +18361,14 @@
           "stability": "stable"
         }
       },
-      "description": "Deletes a repository.",
+      "description": "Delete snapshot repositories.\nWhen a repository is unregistered, Elasticsearch only removes the reference to the location where the repository is storing the snapshots.\nThe snapshots themselves are left untouched and in place.",
```
Comment (karenzone) [inline output/schema/schema.json:18364]:
```suggestion
      "description": "Delete snapshot repositories.\nWhen a repository is unregistered, Elasticsearch removes only the reference to the location where the repository is storing the snapshots.\nThe snapshots themselves are left untouched and in place.",
```

File: specification/snapshot/delete_repository/SnapshotDeleteRepositoryRequest.ts [hunk 4]
```diff
@@ -22,9 +22,13 @@ import { Names } from '@_types/common'
 import { Duration } from '@_types/Time'
 
 /**
+ * Delete snapshot repositories.
+ * When a repository is unregistered, Elasticsearch only removes the reference to the location where the repository is storing the snapshots.
```
Comment (karenzone) [inline specification/snapshot/delete_repository/SnapshotDeleteRepositoryRequest.ts:26]:
```suggestion
 * When a repository is unregistered, Elasticsearch removes only the reference to the location where the repository is storing the snapshots.
```

File: specification/snapshot/restore/SnapshotRestoreRequest.ts [hunk 5]
```diff
@@ -23,9 +23,29 @@ import { Indices, Name } from '@_types/common'
 import { Duration } from '@_types/Time'
 
 /**
+ * Restore a snapshot.
+ * Restore a snapshot of a cluster or data streams and indices.
+ *
+ * You can only restore a snapshot to a running cluster with an elected master node.
```
Comment (karenzone) [inline specification/snapshot/restore/SnapshotRestoreRequest.ts:29]:
```suggestion
 * You can restore a snapshot only to a running cluster with an elected master node.
```

**Review by karenzone — APPROVED** [review]
Left some suggestions inline wrt moving the word "only" closer to the restriction it modifies. I know this was existing content, so you can decide. Otherwise, LGTM. 🚀

File: output/openapi/elasticsearch-openapi.json [hunk 6]
```diff
@@ -32027,7 +32048,8 @@
         "tags": [
           "snapshot"
         ],
-        "summary": "Deletes a repository",
+        "summary": "Delete snapshot repositories",
+        "description": "When a repository is unregistered, Elasticsearch only removes the reference to the location where the repository is storing the snapshots.\nThe snapshots themselves are left untouched and in place.",
```
[the remark this answers is not in the document]
Comment (lcawl) [reply output/openapi/elasticsearch-openapi.json]: Fixed in https://github.com/elastic/elasticsearch-specification/pull/3314/commits/dc70256ca197d43bd3796691a02dc19e2a9a13f7

File: output/openapi/elasticsearch-openapi.json (hunk 1 above)
Comment (lcawl) [reply output/openapi/elasticsearch-openapi.json]: Fixed in https://github.com/elastic/elasticsearch-specification/pull/3314/commits/dc70256ca197d43bd3796691a02dc19e2a9a13f7

File: output/schema/schema.json (hunk 3 above)
Comment (lcawl) [reply output/schema/schema.json]: Fixed in https://github.com/elastic/elasticsearch-specification/pull/3314/commits/dc70256ca197d43bd3796691a02dc19e2a9a13f7

File: specification/snapshot/delete_repository/SnapshotDeleteRepositoryRequest.ts (hunk 4 above)
Comment (lcawl) [reply specification/snapshot/delete_repository/SnapshotDeleteRepositoryRequest.ts]: Fixed in https://github.com/elastic/elasticsearch-specification/pull/3314/commits/dc70256ca197d43bd3796691a02dc19e2a9a13f7

File: output/openapi/elasticsearch-openapi.json (hunk 2 above)
Comment (lcawl) [reply output/openapi/elasticsearch-openapi.json]: Fixed in https://github.com/elastic/elasticsearch-specification/pull/3314/commits/dc70256ca197d43bd3796691a02dc19e2a9a13f7

**Comment by lcawl** [comment]
> Left some suggestions inline wrt moving the word "only" closer to the restriction it modifies. I know this was existing content, so you can decide. Otherwise, LGTM. 🚀

Thanks for the edits!  I've made added the suggested changes to the source files and regenerated the output files.

**Pull request merged**